core/src/main/java/org/jboss/sbomer/core/features/sbom/utils/SbomUtils.java

```diff
@@ -600,15 +600,15 @@ public static void updateBomRef(Bom bom, Component component, String oldRef, Str
             if (bom.getDependencies() != null) {
                 List<Dependency> updatedDependencies = new ArrayList<>(bom.getDependencies().size());
                 for (Dependency dependency : bom.getDependencies()) {
-                    updateDependencyRef(dependency, oldRef, newRef);
-                    updatedDependencies.add(dependency);
+                    Dependency updatedDependency = updateDependencyRef(dependency, oldRef, newRef);
+                    updatedDependencies.add(updatedDependency);
                 }
                 bom.setDependencies(updatedDependencies);
             }
         }
     }
 
-    public static void updateDependencyRef(Dependency dependency, String oldRef, String newRef) {
+    public static Dependency updateDependencyRef(Dependency dependency, String oldRef, String newRef) {
```

Comment:
Huh, do I understand it correctly, that before the method was actually not doing anything?
Created an updated dependency, that it throw away at the end?